fetchai/ledger/api/token.py

```diff
@@ -257,22 +228,117 @@ def collect_stake(self, entity: Entity, fee: int):
         :raises: ApiError on any failures
         """
+
         ENDPOINT = 'collectStake'
 
-        # format the data to be closed by the transaction
+        tx = TokenTxFactory.collect_stake(entity, fee)
+        self._set_validity_period(tx)
 
-        # wildcard for the moment
-        shard_mask = BitVector()
+        # encode and sign the transaction
+        encoded_tx = transaction.encode_transaction(tx, [entity])
+
+        # submit the transaction
+        return self._post_tx_json(encoded_tx, ENDPOINT)
+
+
+class TokenTxFactory(TransactionFactory):
+    API_PREFIX = 'fetch.token'
+
+    @classmethod
+    def wealth(cls, entity: Entity, amount: int):
 
         # build up the basic transaction information
-        tx = self._create_skeleton_tx(fee)
-        tx.from_address = Address(entity)
-        tx.target_chain_code(self.API_PREFIX, shard_mask)
-        tx.action = 'collectStake'
+        tx = cls._create_action_tx(1, entity, 'wealth')
+
         tx.add_signer(entity)
 
-        # encode and sign the transaction
-        encoded_tx = encode_transaction(tx, [entity])
+        # format the transaction payload
+        tx.data = cls._encode_json({
+            'address': entity.public_key,
+            'amount': amount
+        })
 
-        # submit the transaction
-        return self._post_tx_json(encoded_tx, ENDPOINT)
+        return tx
+
+    @classmethod
+    def deed(cls, entity: Entity, deed: Deed, signatories: list = None):
+        tx = cls._create_action_tx(10000, entity, 'deed')
+
+        if signatories:
+            for sig in signatories:
+                tx.add_signer(sig)
+        else:
+            tx.add_signer(entity)
+
+        deed_json = deed.deed_creation_json()
+
+        tx.data = cls._encode_json(deed_json)
+
+        return tx
+
+    @classmethod
+    def transfer(cls, entity: Entity, to: AddressLike, amount: int, fee: int, signatories: list = None):
+
+        # build up the basic transaction information
+        tx = cls._create_skeleton_tx(fee)
+        tx.from_address = Address(entity)
+        tx.add_transfer(to, amount)
+
+        if signatories:
+            for ent in signatories:
+                tx.add_signer(ent)
+        else:
+            tx.add_signer(entity)
+
+        return tx
+
+    @classmethod
+    def add_stake(cls, entity: Entity, amount: int, fee: int, signatories: list = None):
+
+        # build up the basic transaction information
+        tx = cls._create_action_tx(fee, entity, 'addStake')
+        if signatories:
+            for ent in signatories:
+                tx.add_signer(ent)
+        else:
+            tx.add_signer(entity)
+
+        # format the transaction payload
+        tx.data = cls._encode_json({
+            'address': entity.public_key,
+            'amount': amount
+        })
+
+        return tx
+
+    @classmethod
+    def de_stake(cls, entity: Entity, amount: int, fee: int, signatories: list = None):
```

Comment:
Many of these methods, such as destake are now duplicated with other identiacl methods. Kinda this whole thing needs seriously looking at before merging as it will make the codebase hard to understand